src/StreamJsonRpc/NerdbankMessagePackFormatter.FormatterContextBuilder.cs

```diff
@@ -0,0 +1,219 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Immutable;
+using System.IO.Pipelines;
+using Nerdbank.MessagePack;
+using PolyType;
+using PolyType.Abstractions;
+using StreamJsonRpc.Reflection;
+
+namespace StreamJsonRpc;
+
+/// <summary>
+/// Serializes JSON-RPC messages using MessagePack (a fast, compact binary format).
+/// </summary>
+/// <remarks>
+/// The MessagePack implementation used here comes from https://github.com/AArnott/Nerdbank.MessagePack.
+/// </remarks>
+public sealed partial class NerdbankMessagePackFormatter
+{
+    /// <summary>
+    /// Provides methods to build a serialization context for the <see cref="NerdbankMessagePackFormatter"/>.
+    /// </summary>
+    public class FormatterContextBuilder
+    {
+        private readonly NerdbankMessagePackFormatter formatter;
```

Comment:
This is per-connection state. The dependency here needs to be broken to support reusability of the FormatterContext.